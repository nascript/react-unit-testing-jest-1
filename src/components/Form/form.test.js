import React from "react";
import { render } from "@testing-library/react";
import { Form } from "./";

describe("<Form />", () => {
  const user = {
    firstName: "Mo",
    lastName: "Dwina",
    email: "b@b.com",
  };

  test("renders component", () => {
    const { container } = render(
      <Form user={user} handleSubmit={() => {}} handleChange={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  test("should disable the button when one of user data is missing", () => {
    const { queryByText } = render(
      <Form user={{}} handleSubmit={() => {}} handleChange={() => {}} />
    );
    const button = queryByText("Submit");
    expect(button.classList).toContain("disabled");
  });

  test("should remove disable class from the button when all user data is provided", () => {
    const { queryByText } = render(
      <Form user={user} handleSubmit={() => {}} handleChange={() => {}} />
    );
    const button = queryByText("Submit");
    expect(button.classList).not.toContain("disabled");
  });
});
