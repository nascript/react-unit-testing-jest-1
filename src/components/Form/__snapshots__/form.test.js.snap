// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders component 1`] = `
<div>
  <div>
    <h1>
      Registration form
    </h1>
    <form>
      <input
        name="firstName"
        placeholder="First Name"
        type="text"
        value="Mo"
      />
      <input
        name="lastName"
        placeholder="Last Name"
        type="text"
        value="Dwina"
      />
      <input
        name="email"
        placeholder="Email"
        type="email"
        value="b@b.com"
      />
      <button
        class=""
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
